### ISSUE TITLE
What is the purpose of app.json file? Is it safe to delete it?

### ISSUE THREAD

**Issue by mtshv** [react-native-community/cli#1113]
`app.json file` is created, when a new react-native project is initiated from the **react-native cli**
It looks, like this:
```
{
  "name": "Name",
  "displayName": "DisplayName"
}
```

I am wondering, what purpose does it have? And is it safe to delete this file?

P.S. I did some googling on this topic but found only [outdated info](https://stackoverflow.com/questions/42409459/react-native-0-41-app-json), like this file, was used for `eject` command that does not exist anymore.

**Comment by brentvatne** [comment]
typically it is imported in index.js to get the name for the root component: https://github.com/facebook/react-native/blob/44beb2a685b7ceb0311bde7d0d33cb70bb891d30/template/index.js#L7

tl;dr: if you just inline the name in index.js or do something equivalent to that then yes you can delete it.

the history of this file:

- [expo](https://expo.io) (formerly exponent) used exp.json as a way to configure things like your app icon, splash screen, and various other project config properties that you would typically configure in plist/xml files or in your IDE.
- the react-native team at facebook and expo worked together to make create-react-native-app (CRNA) in order to lower the barrier to entry to get started with react-native. CRNA used expo tools like expo client, some CLI libraries, and the project configuration format, but we wanted to make sure that CRNA wasn't specifically tied to just expo - so anyone else could build a similar tool and leverage CRNA for it. this motivated us to change exp.json to app.json, and make expo config just a key on that.
- we wanted projects created with CRNA and rn-cli to have a similar layout, so we moved the react-native template to use App.js and a single index.js (rather than index.ios.js and index.android.js iirc).
- we also wanted to be able to transition from a project built with CRNA (which did not have ios and android project directories with native code) towards one with full access to native code, so we added the eject command to rn-cli that would re-create the native project based on some configuration. the configuration was decided to be stored in app.json.
- the hope with app.json was to build tooling to apply config to native projects but it hasn't materialized yet. we've since moved the eject implementation entirely to expo-cli and continued building this there, now on ejecting we support a variety of properties, from facebook and google config to entitlements and permissions and so on ([ios properties](https://github.com/expo/expo-cli/tree/master/packages/config/src/ios) [android properties](https://github.com/expo/expo-cli/tree/master/packages/config/src/android)).

**Comment by mtshv** [comment]
@brentvatne thank you very much for such a great and detailed answer. It was really interesting to understand the history behind this file! 🙏
